drivers/driver_functions.go

```diff
@@ -4,39 +4,20 @@ import (
 	"fmt"
 	"github.com/pkg/errors"
 	"github.com/rancher/go-rancher/v2"
-	"github.com/rancher/rancher-auth-service/util"
+	"github.com/rancher/webhook-service/config"
 	"net/http"
 	"time"
 )
 
 //WebhookDriver interface for all drivers
 type WebhookDriver interface {
-	ConstructPayload(input map[string]interface{}) (map[string]interface{}, error)
-	Execute(payload map[string]interface{}) (int, map[string]interface{}, error)
+	Execute(payload map[string]interface{}, envConfig config.Config) (int, map[string]interface{}, error)
```

Comment:
I think we need to change this to pass the rancher client in as a parameter to Execute. The reasons for this are that
1. We don't want the drivers to have to know the details of how to create clients because this makes drivers harder to implement.
2. We want the framework to be able to manage clients so that we can address the resource leak in a single place.